### ISSUE TITLE
Knex: Timeout acquiring a connection. The pool is probably full

### ISSUE THREAD

**Issue by SamGreenwood1** [louislam/uptime-kuma#5033]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

I noticed that I'm getting a `Knex: Timeout` error. Does anyone know what the error means?

### 📝 Error Message(s) or Log

Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?

### 🐻 Uptime-Kuma Version

1.23.13

### 💻 Operating System and Arch

Ubuntu 20.04 LTS

### 🌐 Browser

Brave 1.68.137, Chromium 127.0.6533.100

### 🖥️ Deployment Environment

- i don't know the exact answers to the first few questions because it's running in Cloudron (v7.6.4) on Linode
- 44 Monitors

**Comment by vishalsabhaya** [comment]
@SamGreenwood1 

there is only 2 possibilities because it’s running on storage.
1) storage is full
2) pool limit reached

If remote storage then might be network have problem.
I don’t think kuma work with network storage.

**Comment by sparker81** [comment]
We're having this issue as well. The storage isn't full, we have checked that. @vishalsabhaya, what pool are we referring to? Is this a SQL term?

**Comment by sparker81** [comment]
> ### ⚠️ Please verify that this question has NOT been raised before.
> * [x]  I checked and didn't find similar issue
> 
> ### 🛡️ Security Policy
> * [x]  I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)
> 
> ### 📝 Describe your problem
> I noticed that I'm getting a `Knex: Timeout` error. Does anyone know what the error means?
> 
> ### 📝 Error Message(s) or Log
> Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
> 
> ### 🐻 Uptime-Kuma Version
> 1.23.13
> 
> ### 💻 Operating System and Arch
> Ubuntu 20.04 LTS
> 
> ### 🌐 Browser
> Brave 1.68.137, Chromium 127.0.6533.100
> 
> ### 🖥️ Deployment Environment
> * i don't know the exact answers to the first few questions because it's running in Cloudron (v7.6.4) on Linode
> * 44 Monitors

Just out of curiosity, what time of the day are you getting this error?

**Comment by CommanderStorm** [comment]
A fix for this is in master, but needs a database migration written to be released as V2.0.
See #4500 for additional recommendations how to get over the performance issue.

(Written on mobile => not writing out my usual copy paste text.. see the issues linking to #4500)

**Comment by CommanderStorm** [comment]
Duplicate of [a lot of other issues](https://github.com/louislam/uptime-kuma/issues?q=is%3Aissue+Timeout+acquiring+a+connection+is%3Aclosed).

I am going to close this as resolved by `v2.0` as I don't see evidence to the contrary.

> TL;DR:
> In v1, reducing either retention/ping-rate/.. or moving to a faster storage solution is the only option to get around this problem.
> In v2, we have likely resolved this problem. See #4500 for further details of what needs to happen before we can release this version.
